test/variable_test.cpp

```diff
@@ -956,3 +956,39 @@ TEST(VariableSlice, slice_binary_operations) {
   EXPECT_TRUE(equals(difference.get<const Data::Value>(), {-1, -1}));
   EXPECT_TRUE(equals(product.get<const Data::Value>(), {2, 12}));
 }
+
+TEST(Variable, reshape) {
+  Variable var(Data::Value{}, {{Dim::X, 2}, {Dim::Y, 3}}, {1, 2, 3, 4, 5, 6});
+  auto view = var.reshape({Dim::Row, 6});
+  ASSERT_EQ(view.size(), 6);
+  ASSERT_EQ(view.dimensions(), Dimensions({Dim::Row, 6}));
+  EXPECT_TRUE(equals(view.get<const Data::Value>(), {1, 2, 3, 4, 5, 6}));
+
+  auto view2 = var.reshape({{Dim::Row, 3}, {Dim::Z, 2}});
+  ASSERT_EQ(view2.size(), 6);
+  ASSERT_EQ(view2.dimensions(), Dimensions({{Dim::Row, 3}, {Dim::Z, 2}}));
+  EXPECT_TRUE(equals(view2.get<const Data::Value>(), {1, 2, 3, 4, 5, 6}));
+}
+
+TEST(Variable, reshape_fail) {
+  Variable var(Data::Value{}, {{Dim::X, 2}, {Dim::Y, 3}}, {1, 2, 3, 4, 5, 6});
+  EXPECT_THROW_MSG(var.reshape({Dim::Row, 5}), std::runtime_error,
+                   "Cannot reshape to dimensions with different volume");
+}
+
+TEST(Variable, reshape_and_slice) {
+  Variable var(Data::Value{}, {Dim::Spectrum, 16},
+               {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16});
+
+  auto slice =
+      var.reshape({{Dim::X, 4}, {Dim::Y, 4}})(Dim::X, 1, 3)(Dim::Y, 1, 3);
+  EXPECT_TRUE(equals(slice.get<const Data::Value>(), {6, 7, 10, 11}));
+
+  Variable center =
+      var.reshape({{Dim::X, 4}, {Dim::Y, 4}})(Dim::X, 1, 3)(Dim::Y, 1, 3)
```

Comment:
Can we avoid a syntax reshape(...)(...)(...) here? For example, provide slice(...) function in addition to the slice operator().

Reply:
Sure that could be added.